arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -518,6 +518,56 @@ static void initializeArrowState(@TempDir Path sharedTempDir) throws IOException
         .explainContains(plan);
   }
 
+  @Test void testDecimalProject() {
+    String sql = "select SAL from EMP";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowProject(SAL=[$5])\n"
+        + "    ArrowTableScan(table=[[ARROW, EMP]], fields=[[0, 1, 2, 3, 4, 5, 6, 7]])\n\n";
+    String result = "SAL=800.00\n"
+        + "SAL=1600.00\n"
+        + "SAL=1250.00\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .limit(3)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  @Test void testCastDecimalToSmallInt() {
```

Comment:
If you really want to test the functionality of the cast you should test a case where the value is too large for the target type.

Reply:
This will fail. I tried it. I think I should think about how to show this failure case in the test.

Reply:
```
  @Test void testCastDecimalToTinyInt() {
    String sql = "select CAST(LOSAL AS TINYINT) from SALGRADE";
    String plan =
        "PLAN=EnumerableCalc(expr#0..2=[{inputs}], expr#3=[CAST($t1):TINYINT], EXPR$0=[$t3])\n"
            + "  ArrowToEnumerableConverter\n"
            + "    ArrowTableScan(table=[[ARROW, SALGRADE]], fields=[[0, 1, 2]])\n\n";
    String result = "EXPR$0=700\n";

    CalciteAssert.that()
        .with(arrow)
        .query(sql)
        .typeIs("[EXPR$0 TINYINT]")
        .limit(1)
        .returns(result)
        .explainContains(plan);
  }
```
This test will fail, but I don't know how to describe it with a test
The arrow adapter still has a lot of problems, I'm not sure if adding these tests will be useful
I borrowed from: https://github.com/apache/datafusion-comet

Reply:
You are testing a feature which does not really work.
Is there an issue for the bug? If there is you can add the negative test and disable it until the bug is fixed.